src/Analyzers/CSharp/Analyzers/SimplifyLinqExpression/CSharpSimplifyLinqExpressionDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,179 @@
+            var linqMethods = ImmutableHashSet.CreateBuilder<IMethodSymbol>();
+
+            var fullyQualifiedEnumerableName = typeof(Enumerable)?.FullName;
+            if (fullyQualifiedEnumerableName is null)
+            {
+                return;
+            }
+
+            var enumerableNamedType = context.Compilation.GetTypeByMetadataName(fullyQualifiedEnumerableName);
+            if (enumerableNamedType is null)
+            {
+                return;
+            }
+
+            var enumerableMethods = enumerableNamedType.GetMembers(nameof(Enumerable.Where)).OfType<IMethodSymbol>();
+            whereMethods.UnionWith(enumerableMethods);
+
+            // add all valid linq calls
+            foreach (var id in _nonEnumerableReturningLinqMethodNames)
+            {
+                enumerableMethods = enumerableNamedType.GetMembers(id).OfType<IMethodSymbol>();
+                linqMethods.UnionWith(enumerableMethods);
+            }
+
+            if (whereMethods.Count > 0 && linqMethods.Count > 0)
+            {
+                context.RegisterOperationAction(
+                    context => AnalyzeAction(context, whereMethods.ToImmutable(),
+                    linqMethods.ToImmutable()),
+                    OperationKind.Invocation);
+            }
+        }
+
+        public void AnalyzeAction(OperationAnalysisContext context, ImmutableHashSet<IMethodSymbol> whereMethods, ImmutableHashSet<IMethodSymbol> linqMethods)
+        {
+            var invocationOperation = (IInvocationOperation)context.Operation;
+            var semanticModel = context.Operation.SemanticModel;
+            var invocationArgument = invocationOperation.Arguments.FirstOrDefault();
+
+            // Check to make sure the invocation argument is an InvocationExpressionSyntax
+            // Example: If invocationOperation syntax is Data().Where(...).Single(), then invocationArgument is Data.Where(...)
+            if (invocationArgument is null)
+            {
+                return;
+            }
+
+            var targetOperation = invocationArgument.Children.FirstOrDefault(c => c is IOperation);
+
+            // Example: If invocation is Data.Where(...), then the TargetMethod would be .Where(...)
+            if (targetOperation is not IInvocationOperation invocation)
+            {
+                return;
+            }
+
+            var targetDefinition = invocation.TargetMethod.OriginalDefinition;
+
+            // True if the invocation.TargetMethod is a call to System.Linq.Enumerable.Where(...)
+            if (!whereMethods.Contains(targetDefinition))
+            {
+                return;
+            }
+
+            // True if the arguments within the .Where(...) method are lambda expressions
+            var isLambda = invocation.Arguments
+                .Any(c => (c.Syntax is ArgumentSyntax argSyntax && argSyntax.Expression.IsAnyLambda()) ||
+                           c.Syntax is LambdaExpressionSyntax);
```

Comment:
expression is too conflusing.  i'm not sure what it's saying here.  please parenthesize accordingly.  also, why do you use .isAnyLambda for one check, but `is LambdaExpression` for the other?